encoding/thrift/observability_test.go

```diff
@@ -54,92 +55,94 @@ const (
 )
 
 func TestThriftExceptionObservability(t *testing.T) {
-	// TODO(apeatsbond): add HTTP test when feature complete.
-
-	t.Run("exception with annotation", func(t *testing.T) {
-		client, observedLogs, clientMetricsRoot, serverMetricsRoot, cleanup := initClientAndServer(t)
-		defer cleanup()
-
-		ctx, cancel := context.WithTimeout(context.Background(), time.Second)
-		defer cancel()
-
-		_, err := client.Call(ctx, _wantExceptionWithCode)
-		require.Error(t, err, "expected call error")
-
-		ex, ok := err.(*test.ExceptionWithCode)
-		require.True(t, ok, "unexpected Thrift exception")
-		assert.Equal(t, _wantExceptionWithCode, ex.Val, "unexpected response")
-
-		t.Run("logs", func(t *testing.T) {
-			wantFields := []zapcore.Field{
-				zap.String("error", "application_error"),
-				zap.String("errorName", "ExceptionWithCode"),
-				zap.String("errorCode", "invalid-argument"),
-				zap.String("appErrorMessage", "ExceptionWithCode{Val: exception with code}"),
-			}
-			assertLogs(t, wantFields, observedLogs.TakeAll())
-		})
-
-		t.Run("metrics", func(t *testing.T) {
-			wantCounters := []counterAssertion{
-				{
-					Name: "caller_failures",
-					Tags: map[string]string{
-						"error":      "invalid-argument",
-						"error_name": "ExceptionWithCode",
+	transports := []string{tchannel.TransportName, http.TransportName}
+
+	for _, trans := range transports {
```

Comment:
note that wrapping the existing logic in the `for-loop` makes this change look bigger than it really is. No core testing logic changed.  Now, we pass a `trans` variable to indicate the transport type to use when building the client and server.

Actual changes are in `initClientAndServer`, `newClient` and `newServer`